arm9/source/utils/gameutil.c

```diff
@@ -1584,6 +1584,53 @@ u32 ExtractCodeFromCxiFile(const char* path, const char* path_out, char* extstr)
     return 0;
 }
 
+u32 CompressCode(const char* path, const char* path_out) {
+    FIL code_file;
+    char dest[256];
+    if (fvx_open(&code_file, path, FA_READ) != FR_OK) return 1;
+    if (!path_out && (fvx_rmkdir(OUTPUT_PATH) != FR_OK)) return 1;
+    strncpy(dest, path_out ? path_out : OUTPUT_PATH, 256);
+    if (!CheckWritePermissions(dest)) return 1;
+    
+    // allocate memory
+    u32 code_dec_size = fvx_size(&code_file);
+    u8* code_dec = (u8*) malloc(code_dec_size);
+    u32 code_cmp_size = code_dec_size;
+    u8* code_cmp = (u8*) malloc(code_cmp_size);
+    if (!code_dec) {
```

Comment:
Not checking `code_cmp` here, also not freeing stuff in case only one got not allocated. Isn't it possible to do that in only one buffer, same as decompression? You will hit memory limits fast this way.